helix-lock/src/main/java/org/apache/helix/lock/DistributedLock.java

```diff
@@ -49,4 +49,9 @@
    * false if the user is not the lock owner or the lock doesn't have a owner
    */
   boolean isCurrentOwner();
+
+  /**
+   * Close the lock and release resources
+   */
+  void close();
```

Comment:
I suggest making this interface AutoClosable, then you implement close() naturally.

Reply:
As discussed offline, to avoid unexpected close of the lock, we will not implement AutoClosable interface for now.